shared/js/background/load.es6.js

```diff
@@ -27,35 +38,60 @@ function returnResponse (xhr, returnType) {
  *  - source: requests are internal by default. set source to 'external' for non-extension URLs
  *  - etag: set an if-none-match header
  */
-function loadExtensionFile (params, cb) {
-    var xhr = new XMLHttpRequest()
+function loadExtensionFile (params) {
+    let xhr = new XMLHttpRequest()
+    let url = params.url
 
     if (params.source === 'external') {
-        xhr.open('GET', params.url)
+        if (dev) {
+            if (url.indexOf('?') > -1) {
+                url += '&'
+            } else {
+                url += '?'
+            }
+
+            url += 'dev=1'
+        }
```

Comment:
Since we are using promises and started doing search parameters manipulation, IMO it would simplify things quite a bit to switch to `fetch` and `URLSearchParams` from `XMLHttpRequest`.